models/user.rb

```diff
@@ -42,7 +42,7 @@ def unfollow(other_user)
   end
 
   def build_timeline()
-    if $redis.exists("timeline:user:#{self.id}")
+    if $redis.ttl("timeline:user:#{self.id}")>0
```

Comment:
No idea. In some examples people were using ttl along side expiration. maybe exists is the better performing one?